graphql/src/index.ts

```diff
@@ -23,47 +22,71 @@ const { ElasticSearchAPI } = require('./datasources/es');
 const OK = 'OK';
 const SERVER_IS_NOT_READY = 'SERVER_IS_NOT_READY';
 
+type UnifiedSearchQuery = {
+  q?: String;
+  ontology?: string;
+  index?: string;
+} & ConnectionArguments;
+
+function edgesFromEsResults(results: any, getCursor: any) {
+  return results.hits.hits.map(function (
+    e: { _score: any; _source: { venue: any } },
+    index: number
+  ) {
+    return {
+      cursor: getCursor(index + 1),
+      node: {
+        _score: e._score,
+        venue: { venue: e._source.venue }, // pass parent to child resolver. How to do this better?
+      },
+    };
+  });
+}
+
+function getHits(results: any) {
+  return results.hits.total.value;
+}
+
 const resolvers = {
   Query: {
-    unifiedSearch: async (_source: any, { q, ontology, index }: any, { dataSources }: any) => {
-      const res = await dataSources.elasticSearchAPI.getQueryResults(q, ontology, index);
-      return { es_results: res };
+    unifiedSearch: async (
+      _source: any,
+      { q, ontology, index, before, after, first, last }: UnifiedSearchQuery,
+      { dataSources }: any
+    ) => {
+      const connectionArguments = { before, after, first, last };
+
+      const res = await dataSources.elasticSearchAPI.getQueryResults(
+        q,
+        ontology,
+        index,
+        connectionArguments
+      );
+      const results = await res[0];
```

Comment:
I'm not sure how these resolvers work under the hood. In other words, I wonder if it would be better to not resolve the promise on this level and instead do it later on in the resolvers. For now I adopted the simplest approach: resolve here so that shared fields can be found out only once.